src/components/hitbox.rs

```diff
@@ -0,0 +1,39 @@
+use amethyst::{
+    core::transform::Transform,
+    ecs::prelude::{Component, DenseVecStorage},
+};
+
+pub struct Hitbox2DComponent {
+    pub width: f32,
+    pub height: f32,
+    pub offset_x: f32,
+    pub offset_y: f32,
+    pub offset_rotation: f32, // offset in radians
+}
+
+impl Component for Hitbox2DComponent {
+    type Storage = DenseVecStorage<Self>;
+}
+
+impl Hitbox2DComponent {
+    pub fn is_colliding(
+        &self,
+        transform_a: &Transform,
+        hitbox_b: &Hitbox2DComponent,
+        transform_b: &Transform,
+    ) -> bool {
```

Comment:
nit: could we order the `transform_a` and `transform_b` parameters to be next to each other? 

```suggestion
    pub fn is_colliding(
        &self,
        transform_a: &Transform,
        transform_b: &Transform,
        hitbox_b: &Hitbox2DComponent,
    ) -> bool {
```

Reply:
Since `&self` refers to the hitbox that the function is being called from, I think it makes the most sense to arrange the parameters `&self, hitbox_b, transform_a, transform_b`.